File: backend/app/schemas/alerts.py
from pydantic import BaseModel


class AlertRead(BaseModel):
    kind: str
    key: str
    title: str
    detail: str
    target_type: str | None
    target_id: int | None
    acknowledgeable: bool
    acknowledged: bool


class AlertsResponse(BaseModel):
    alerts: list[AlertRead]
    # Excludes already-acknowledged update alerts — matches typical
    # bell-badge semantics (a badge count an admin has already
    # acknowledged shouldn't keep demanding attention).
    active_count: int

File: frontend/src/lib/api/alerts.ts
import { apiFetch } from "../apiClient";

export interface Alert {
  kind: string;
  key: string;
  title: string;
  detail: string;
  target_type: string | null;
  target_id: number | null;
  acknowledgeable: boolean;
  acknowledged: boolean;
}

export interface AlertsResponse {
  alerts: Alert[];
  active_count: number;
}

export function fetchAlerts(): Promise<AlertsResponse> {
  return apiFetch<AlertsResponse>("/api/alerts");
}

export function acknowledgeAlert(kind: string): Promise<void> {
  return apiFetch<void>(`/api/alerts/${encodeURIComponent(kind)}/acknowledge`, { method: "POST" });
}
